src/nvim/option.c

```diff
@@ -640,43 +618,29 @@ void set_init_3(void)
   char *p = (char *)invocation_path_tail(p_sh, &len);
   p = xmemdupz(p, len);
 
-  {
-    //
-    // Default for p_sp is "| tee", for p_srr is ">".
-    // For known shells it is changed here to include stderr.
-    //
-    if (path_fnamecmp(p, "csh") == 0
-        || path_fnamecmp(p, "tcsh") == 0) {
-      if (do_sp) {
-        p_sp = "|& tee";
-        options[kOptShellpipe].def_val.string = p_sp;
-      }
-      if (do_srr) {
-        p_srr = ">&";
-        options[kOptShellredir].def_val.string = p_srr;
-      }
-    } else if (path_fnamecmp(p, "sh") == 0
-               || path_fnamecmp(p, "ksh") == 0
-               || path_fnamecmp(p, "mksh") == 0
-               || path_fnamecmp(p, "pdksh") == 0
-               || path_fnamecmp(p, "zsh") == 0
-               || path_fnamecmp(p, "zsh-beta") == 0
-               || path_fnamecmp(p, "bash") == 0
-               || path_fnamecmp(p, "fish") == 0
-               || path_fnamecmp(p, "ash") == 0
-               || path_fnamecmp(p, "dash") == 0) {
-      // Always use POSIX shell style redirection if we reach this
-      if (do_sp) {
-        p_sp = "2>&1| tee";
-        options[kOptShellpipe].def_val.string = p_sp;
-      }
-      if (do_srr) {
-        p_srr = ">%s 2>&1";
-        options[kOptShellredir].def_val.string = p_srr;
-      }
+  bool is_csh = path_fnamecmp(p, "csh") == 0 || path_fnamecmp(p, "tcsh") == 0;
+  bool is_known_shell = path_fnamecmp(p, "sh") == 0 || path_fnamecmp(p, "ksh") == 0
+                        || path_fnamecmp(p, "mksh") == 0 || path_fnamecmp(p, "pdksh") == 0
+                        || path_fnamecmp(p, "zsh") == 0 || path_fnamecmp(p, "zsh-beta") == 0
+                        || path_fnamecmp(p, "bash") == 0 || path_fnamecmp(p, "fish") == 0
```

Comment:
Oh sorry I missed this for some reason

Yes it's cosmetic, but in this case it felt necessary to me as the code was borderline impossible to decipher otherwise